### ISSUE TITLE
Dashboards contain analyzer IDs instead of correct names

### ISSUE THREAD

**Issue by crackytsi** [TheHive-Project/TheHive#608]
### Request Type
Bug

### Work Environment

| Question              | Answer
|---------------------------|--------------------
| OS version (server)       | Debian
| OS version (client)       | Seven using Chrome Browser
| TheHive version / git hash   | 3.0.10
| Package Type              | DEB

### Problem Description
The reports contains IDs instead of the correct names e.g. Standard-Report Job-Statistics of Cortex-Analyzers.

**Comment by saadkadhi** [comment]
@crackytsi can you confirm that you are encountering this issue with 3.0.10 and not 3.0.9 and can you provide screenshots? 

I noticed this issue in mini-reports with 3.0.9 and I believe we corrected it through #586. I've run several analysis jobs since then and I cannot reproduce the issue:

![screen shot 2018-06-12 at 05 55 54](https://user-images.githubusercontent.com/2852427/41269519-cdbdc290-6e05-11e8-9115-6ea180d03081.png)
![screen shot 2018-06-12 at 05 55 34](https://user-images.githubusercontent.com/2852427/41269521-cdd59ffa-6e05-11e8-8855-6f5a01cf39e1.png)
![screen shot 2018-06-12 at 05 55 17](https://user-images.githubusercontent.com/2852427/41269522-cdebf516-6e05-11e8-9174-215555f32e02.png)
![screen shot 2018-06-12 at 05 55 09](https://user-images.githubusercontent.com/2852427/41269523-ce01c40e-6e05-11e8-8d0b-ab47460fc119.png)

**Comment by crackytsi** [comment]
Oh not report. I mean dashboards...

**Comment by crackytsi** [comment]
![dashboard](https://user-images.githubusercontent.com/11610825/41272693-af5e6f8e-6e05-11e8-8766-b4f0003dc090.png)

**Comment by 3c7** [comment]
Edited the issue title.

**Comment by saadkadhi** [comment]
@crackytsi  I still cannot reproduce the issue. If the aggregation field is `AnalyzerName` or `AnalyzerDefinition` (these are basically the same), then you'll have the names on 3.0.10:

![screen shot 2018-06-12 at 10 35 31](https://user-images.githubusercontent.com/2852427/41279362-93b96640-6e2c-11e8-9bad-427bfebec30e.png)

If you are using `AnalyzerID` as an aggregation field then you'll get the ID as in your screenshot.

Are you, by any chance running custom/private analyzers? If so, can you provide us (on support@ for ex.) anonymized JSON def files and complementary info such as sample analysis results (anonymized)?

**Comment by crackytsi** [comment]
Strange, on all Systems I see the wrong names. All Systems have 3.0.10 Version.
Can I do some curl commands to retrieve and provide debug information?
For me it seems as the old names were correct and the new ones come from newer Report.

The screenshot is from a Standard Dashboard, I haven't changed anything on it.

**Comment by crackytsi** [comment]
here a sample json Definition. 
```
{
    "name": "AADB",
    "version": "1.0",
    "author": "crackytsi",
    "url": "https://AADB.local",
    "license": "AGPL-V3",
    "baseConfig": "AADB",
    "config": {
        "service": ""
    },
    "description": "AADB Analyzer",
    "dataTypeList": ["ip","domain","fqdn"],
    "command": "AADB/AADB.py"
}

```

**Comment by crackytsi** [comment]
Strange is also: If I select in Dashboard "all time", I see some correct Analyzer names in "Analyzer history" e.g. Hippomore, OTXQuery, MaxMind Geo IP etc. AND some IDs.
If I select "last 30 days", I see only Analyzer IDs, there are no names.

**Comment by nadouani** [comment]
Well, don't forget that when we migrate from Cortex 1 to Cortex 2, the AnalyzerID was the full analyzer name (no persistence) and with Cortex 2, analyer ids are real ids as stored on the DB. So, you just need to update the widget definition and use the AnalyzerName instead of AnalyzerID.

We didn't provide an automatic way to migrate existing dashboards that refer to obsolete fields like `AnalyzerID`, so this needs to be fixed manually

**Comment by crackytsi** [comment]
Hmm. If I Change it, from AnalyzerID to AnalyzerName the widget Shows always "Failed to fetch data, please edit the widget Definition".
Actually there are not much Options or Parameter that can be changed...
Seems to be a bug for me...

**Comment by nadouani** [comment]
I'll take a look on this one. The stats API is probably complaining because of an Elasticsearch mapping on `AnalyzerName` field. We will investigate

**Comment by nadouani** [comment]
This looks working in 3.1.0 which is under dev... need to check it fails on 3.0.x

**Comment by nadouani** [comment]
```
Caused by: java.lang.IllegalArgumentException: Fielddata is disabled on text fields by default. Set fielddata=true on [analyze
rName] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memo
ry. Alternatively use a keyword field instead.
```

In fact, the ES mapping of the `analyzerName` doesn't allow aggregation. This will be fixed in 3.1.0 that will include a DB migration

**Comment by nadouani** [comment]
```
Caused by: java.lang.IllegalArgumentException: Fielddata is disabled on text fields by default. Set fielddata=true on [analyze
rName] in order to load fielddata in memory by uninverting the inverted index. Note that this can however use significant memo
ry. Alternatively use a keyword field instead.
```

In fact, the ES mapping of the `analyzerName` doesn't allow aggregation. This will be fixed in 3.1.0 that will include a DB migration

**Comment by To-om** [comment]
This will be solved after a migration of data.
